python/pyspark/sql/tests.py

```diff
@@ -4550,6 +4550,35 @@ def test_timestamp_dst(self):
         self.assertPandasEqual(pdf, df_from_pandas.toPandas())
 
 
+@unittest.skipIf(
+    not _have_pandas or not _have_pyarrow,
+    _pandas_requirement_message or _pyarrow_requirement_message)
+class MaxResultArrowTests(unittest.TestCase):
+    # These tests are separate as 'spark.driver.maxResultSize' configuration
+    # is a static configuration to Spark context.
+
+    @classmethod
+    def setUpClass(cls):
+        cls.spark = SparkSession.builder \
+            .master("local[4]") \
+            .appName(cls.__name__) \
+            .config("spark.driver.maxResultSize", "10k") \
+            .getOrCreate()
+
+        # Explicitly enable Arrow and disable fallback.
```

Comment:
just to double-check, this test fails even if we do not set these 2 configs, right?

Reply:
This one configuration only `spark.sql.execution.arrow.enabled` because the partial results are being produced from Arrow optimized code path.

`spark.sql.execution.arrow.fallback.enabled` is just to make sure we only test Arrow optimized code path.